README.md
# sped-nfse-fiorilli

## BETHA-TESTS

## Este pacote atende o provedor FIORILLI

[![Build Status][ico-travis]][link-travis]
[![Coverage Status][ico-scrutinizer]][link-scrutinizer]
[![Quality Score][ico-code-quality]][link-code-quality]

[![Latest Stable Version][ico-stable]][link-packagist]
[![Latest Version on Packagist][ico-version]][link-packagist]
[![License][ico-license]][link-packagist]
[![Total Downloads][ico-downloads]][link-downloads]

[![Issues][ico-issues]][link-issues]
[![Forks][ico-forks]][link-forks]
[![Stars][ico-stars]][link-stars]


## Municípios (possivelmente) atendidos pelo provedor

> NOTA: Faltam as URLs dos servidores de Produção desses Municipios, quem desejar usar deve informar essa URL para incluirmos no código.

|n|Município|UF|Ibge|URL Produção|
|:---:|:---|:---:|:---:|:---|
|1|Itacoatiara|AM|1301902|none|
|2|Macapá|AP|1600303|[URL Webservice](http://ws.macapa.ap.gov.br/?wsdl)|
|3|Hidrolândia|GO|5209705|none|
|4|Bacabal|MA|2101202|none|
|5|Codó|MA|2103307|none|
|6|Itapecuru Mirim|MA|2105401|none|
|7|Pedreiras|MA|2108207|none|
|8|Santa Inês|MA|2109908|none|
|9|Timon|MA|2112209|none|
|10|Estiva|MG|3124500|none|
|11|Itaú de Minas|MG|3133758|none|
|12|Jacutinga|MG|3134905|none|
|13|Ouro Fino|MG|3146008|none|
|14|Santa Rita do Sapucaí|MG|3159605|none|
|15|Sapucaí-Mirim|MG|3165404|none|
|16|Sapucaí-Mirim|MG|3165404|none|
|17|Amambai|MS|5000609|none|
|18|Chapadão do Sul|MS|5002951|none|
|19|Corumbá|MS|5003207|[URL Webservice](http://nfse.corumba.ms.gov.br:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|20|Dois Irmãos do Buriti|MS|5003488|none|
|21|Nova Alvorada do Sul|MS|5006002|none|
|22|Paranaíba|MS|5006309|none|
|23|Barra do Garças|MT|5101803|none|
|24|Brasnorte|MT|5101902|none|
|25|Jaciara|MT|5104807|[URL Webservice](http://intranet.jaciara.mt.gov.br:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|26|Lambari D'Oeste|MT|5105234|none|
|27|Mirassol D'Oeste|MT|5105622|[URL Webservice](http://186.235.81.12:5663/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|28|Canaã dos Carajás|PA|1502152|none|
|29|Itaituba|PA|1503606|none|
|30|Rio Maria|PA|1506161|[URL Webservice](http://177.75.236.132:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|31|Xinguara|PA|1508407|none|
|32|Floriano|PI|2203909|none|
|33|Parnaíba|PI|2207702|none|
|34|Picos|PI|2208007|none|
|35|Cardoso Moreira|RJ|3301157|none|
|36|Ariquemes|RO|1100023|[URL Webservice](http://170.82.43.84:5660/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|37|Espigão D´Oeste|RO|1100098|none|
|38|Jaru|RO|1100114|none|
|39|Ji-Paraná|RO|1100122|[URL Webservice](http://177.124.184.59:5660/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|40|Machadinho D´Oeste|RO|1100130|none|
|41|Ouro Preto do Oeste|RO|1100155|none|
|42|Quatro Irmãos|RS|4315313|none|
|43|Marau|RS|4311809|[URL Webservice](http://nfse.pmmarau.com.br:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|44|Serafina Corrêa|RS|4320404|none|
|45|Três Arroios|RS|4321634|[URL Webservice](http://portal.pmtresarroios.com.br:8888/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|46|Altinópolis|SP|3501004|none|
|47|Alvinlândia|SP|3501509|none|
|48|Américo de Campos|SP|3501806|[URL Webservice](http://189.108.31.114:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|49|Araçoiaba da Serra|SP|3502903|none|
|50|Assis|SP|3504008|[URL Webservice](https://nfsews.assis.sp.gov.br/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|51|Bady Bassitt|SP|3504602|none|
|52|Bálsamo|SP|3504800|none|
|53|Bariri|SP|3505203|[URL Webservice](http://sipweb.bariri.sp.gov.br:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|54|Brodowski|SP|3507803|none|
|55|Cabrália Paulista|SP|3508306|none|
|56|Cajobi|SP|3509304|none|
|57|Cândido Rodrigues|SP|3510104|none|
|58|Capela do Alto|SP|3510302|none|
|59|Cedral|SP|3511300|none|
|60|Colômbia|SP|3512100|none|
|61|Cosmorama|SP|3512902|none|
|62|Dracena|SP|3514403|[URL Webservice](http://187.17.201.221:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|63|Duartina|SP|3514502|[URL Webservice](http://186.224.0.62:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|64|Fernando Prestes|SP|3515608|none|
|65|Fernandópolis|SP|3515509|[URL Webservice](http://servicos.fernandopolis.sp.gov.br:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|66|Garça|SP|3516705|none|
|67|Guaiçara|SP|3517208|none|
|68|Guaimbê|SP|3517307|none|
|69|Guaíra|SP|3517406|[URL Webservice](http://186.211.16.210:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|70|Guará|SP|3517703|none|
|71|Guararapes|SP|3518206|none|
|72|Holambra|SP|3519055|none|
|73|Ibaté|SP|3519303|[URL Webservice](http://200.205.131.110:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|74|Ibitinga|SP|3519600|none|
|75|Igarapava|SP|3520103|[URL Webservice](http://138.117.189.204:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|76|Indiaporã|SP|3520707|none|
|77|Inúbia Paulista|SP|3520806|none|
|78|Itajobi|SP|3521903|none|
|79|José Bonifácio|SP|3525706|none|
|80|Junqueirópolis|SP|3526001|none|
|81|Laranjal Paulista|SP|3526407|none|
|82|Martinópolis|SP|3529203|none|
|83|Miguelópolis|SP|3529708|none|
|84|Mirandópolis|SP|3530102|none|
|85|Mirassolândia|SP|3530409|none|
|86|Mongaguá|SP|3531100|none|
|87|Monte Aprazível|SP|3531407|none|
|88|Morungaba|SP|3532009|[URL Webservice](http://servicos.morungaba.sp.gov.br:5600/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|89|Nova Aliança|SP|3532801|none|
|90|Nova Granada|SP|3533007|none|
|91|Novo Horizonte|SP|3533502|[URL Webservice](http://nfews.novohorizonte.sp.gov.br/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|92|Palmares Paulista|SP|3535101|none|
|93|Panorama|SP|3535408|none|
|94|Patrocínio Paulista|SP|3536307|none|
|95|Paulo de Faria|SP|3536604|none|
|96|Pedreira|SP|3537107|none|
|97|Pirajuí|SP|3538907|[URL Webservice](http://189.57.88.130:3392/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|98|Pirangi|SP|3539004|[URL Webservice](http://portal.pmpirangi.com.br:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|99|Pirapozinho|SP|3539202|none|
|100|Pontal|SP|3540200|none|
|101|Potirendaba|SP|3540804|none|
|102|Promissão|SP|3541604|none|
|103|Quatá|SP|3541703|none|
|104|Queiroz|SP|3541802|none|
|105|Ribeirão Bonito|SP|3542909|none|
|106|Ribeirão Corrente|SP|3543105|none|
|107|Rio das Pedras|SP|3544004|[URL Webservice](http://riodaspedras.brgov.com.br:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|108|Riolândia|SP|3544202|none|
|109|Sabino|SP|3544608|none|
|110|Sales|SP|3544806|none|
|111|Sales Oliveira|SP|3544905|[URL Webservice](http://issweb.salesoliveira.sp.gov.br:9090/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|112|Salto de Pirapora|SP|3545308|none|
|113|Santo Anastácio|SP|3547700|none|
|114|Santo Antônio de Posse|SP|3548005|[URL Webservice](http://177.36.128.42:8080/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|115|São Joaquim da Barra|SP|3549409|[URL Webservice](http://187.72.128.113:5661/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|116|Serra Azul|SP|3551405|none|
|117|Serra Negra|SP|3551603|[URL Webservice](http://intra.serranegra.sp.gov.br/IssWeb-ejb/IssWebWS/IssWebWS?wsdl)|
|118|Sud Mennucci|SP|3552304|none|
|119|Ubarana|SP|3555356|none|
|120|Urânia|SP|3555802|none|


## Observações

- Recomenda-se a utilização do envio síncrono, pois as notas enviadas no envio assíncrono são processadas somente as 2h da manhã de cada dia;
- É obrigatório a informação de login e senha para todos os métodos de comunicação.
- Para a emissão das notas no ambiente de homologação deve ser utilizado os dados de um usuário de testes liberado pela prefeitura: CNPJ=01001001000113, IM:15000, Login=01001001000113, Senha=123456, cMun: 3504800, aliquta: 2%, codigo do serviço: 01.05 , ExigibilidadeISS: 1 – Exigível;  RegimeEspecialTributacao: vazio, OptanteSimplesNacional: 2 – Não, IncentivoFiscal: 2 – Não;
- No cancelamento, pode-se informar o parâmetro CodigoCancelamento. Se este não for informado, o componente assumirá este valor como 2 (Serviço não prestado).
- No cancelamento em Homologação, o parâmetro extra CodigoMunicipio a ser informado deverá ser 3504800, pois o servidor de Homologação é situado em Bálsamo - SP;
- Através do site de NFSe disponibilizado pela prefeitura da cidade é possível emitir notas com mais de um item, porém pelo WebService não é possível de acordo com o schemas.xsd. Ao consultar notas emitidas com mais de um item é retornado apenas o último item da nota. Devido a isso não é possível emitir e imprimir notas com mais de um item.

- O código do serviço segue o que está descrito na LC-116.
- O padrão em questão possui uma formatação diferente para o código de tributação do município. O formato é o código da atividade (000010) + código atividade desdobro (0000005). Ess informação pode ser encontrada no portal da cidade, na opção "Consultar alíquotas".
- O código CNAE não tem nenhuma informação específica, mas também não é obrigatório. 

## Dependências

- PHP >= 7.1
- ext-curl
- ext-soap
- ext-zlib
- ext-dom
- ext-openssl
- ext-json
- ext-simplexml
- ext-libxml

### Outras Libs

- nfephp-org/sped-common
- justinrainbow/json-schema


## Contribuindo
Este é um projeto totalmente *OpenSource*, para usa-lo e modifica-lo você não paga absolutamente nada. Porém para continuarmos a mante-lo é necessário qua alguma contribuição seja feita, seja auxiliando na codificação, na documentação ou na realização de testes e identificação de falhas e BUGs.

**Este pacote está listado no [Packgist](https://packagist.org/)**

*Durante a fase de desenvolvimento e testes este pacote deve ser instalado com:*
```bash
composer require nfephp-org/sped-nfse-fiorilli:dev-master
```

*Ou ainda,*
```bash
composer require nfephp-org/sped-nfse-fiorilli:dev-master --prefer-dist
```

*Ou ainda alterando o composer.json do seu aplicativo inserindo:*
```json
"require": {
    "nfephp-org/sped-nfse-fiorilli" : "dev-master"
}
```

> NOTA: Ao utilizar este pacote ainda na fase de desenvolvimento não se esqueça de alterar o composer.json da sua aplicação para aceitar pacotes em desenvolvimento, alterando a propriedade "minimum-stability" de "stable" para "dev".
> ```json
> "minimum-stability": "dev",
> "prefer-stable": true
> ```

*Após os stable realeases estarem disponíveis, este pacote poderá ser instalado com:*
```bash
composer require nfephp-org/sped-nfse-fiorilli
```
Ou ainda alterando o composer.json do seu aplicativo inserindo:
```json
"require": {
    "nfephp-org/sped-sped-nfse-fiorilli" : "^1.0"
}
```

## Forma de uso
vide a pasta *Examples*

## Log de mudanças e versões
Acompanhe o [CHANGELOG](CHANGELOG.md) para maiores informações sobre as alterações recentes.

## Testing

Todos os testes são desenvolvidos para operar com o PHPUNIT

## Security

Caso você encontre algum problema relativo a segurança, por favor envie um email diretamente aos mantenedores do pacote ao invés de abrir um ISSUE.

## Credits

Roberto L. Machado (owner and developer)

## License

Este pacote está diponibilizado sob LGPLv3 ou MIT License (MIT). Leia  [Arquivo de Licença](LICENSE.md) para maiores informações.

[ico-stable]: https://poser.pugx.org/nfephp-org/sped-nfse-fiorilli/version
[ico-stars]: https://img.shields.io/github/stars/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-forks]: https://img.shields.io/github/forks/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-issues]: https://img.shields.io/github/issues/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-travis]: https://img.shields.io/travis/nfephp-org/sped-nfse-fiorilli/master.svg?style=flat-square
[ico-scrutinizer]: https://img.shields.io/scrutinizer/coverage/g/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-code-quality]: https://img.shields.io/scrutinizer/g/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-downloads]: https://img.shields.io/packagist/dt/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-version]: https://img.shields.io/packagist/v/nfephp-org/sped-nfse-fiorilli.svg?style=flat-square
[ico-license]: https://poser.pugx.org/nfephp-org/sped-nfse-fiorilli/license?style=flat-square


[link-packagist]: https://packagist.org/packages/nfephp-org/sped-nfse-fiorilli
[link-travis]: https://travis-ci.org/nfephp-org/sped-nfse-fiorilli
[link-scrutinizer]: https://scrutinizer-ci.com/g/nfephp-org/sped-nfse-fiorilli/code-structure
[link-code-quality]: https://scrutinizer-ci.com/g/nfephp-org/sped-nfse-fiorilli
[link-downloads]: https://packagist.org/packages/nfephp-org/sped-nfse-fiorilli
[link-author]: https://github.com/nfephp-org
[link-issues]: https://github.com/nfephp-org/sped-nfse-fiorilli/issues
[link-forks]: https://github.com/nfephp-org/sped-nfse-fiorilli/network
[link-stars]: https://github.com/nfephp-org/sped-nfse-fiorilli/stargazers



CHANGELOG.md
# Changelog

All Notable changes to `sped-nfse-fiorilli` will be documented in this file.

Todas as atualizações a partir de 31/10/2018 devem observar os princípios [Mantendo o CHANGELOG](http://keepachangelog.com/).

## 1.0.0-dev

## Notas da versão:

### Added
- Nothing

### Deprecated
- Nothing

### Fixed
- Nothing

### Removed
- Nothing

### Security
- Nothing


examples/geraURLjson.php
<?php

$urlhom = "http://fi1.fiorilli.com.br:5663/IssWeb-ejb/IssWebWS/IssWebWS";
$urlprod = "";

$ns = "http://www.abrasf.org.br/nfse.xsd";
$soapns = "http://ws.issweb.fiorilli.com.br/";

$muns = [
    ['ibge' => '1301902', 'mun' => 'Itacoatiara', 'uf' => 'AM', 'url' => ''],
    ['ibge' => '1600303', 'mun' => 'Macapá', 'uf' => 'AP', 'url' => 'http://ws.macapa.ap.gov.br'],
    ['ibge' => '5209705', 'mun' => 'Hidrolândia', 'uf' => 'GO', 'url' => ''],
    ['ibge' => '2101202', 'mun' => 'Bacabal', 'uf' => 'MA', 'url' => ''],
    ['ibge' => '2103307', 'mun' => 'Codó', 'uf' => 'MA', 'url' => ''],
    ['ibge' => '2105401', 'mun' => 'Itapecuru Mirim', 'uf' => 'MA', 'url' => ''],
    ['ibge' => '2108207', 'mun' => 'Pedreiras', 'uf' => 'MA', 'url' => ''],
    ['ibge' => '2109908', 'mun' => 'Santa Inês', 'uf' => 'MA', 'url' => ''],
    ['ibge' => '2112209', 'mun' => 'Timon', 'uf' => 'MA', 'url' => ''],
    ['ibge' => '3124500', 'mun' => 'Estiva', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '3133758', 'mun' => 'Itaú de Minas', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '3134905', 'mun' => 'Jacutinga', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '3146008', 'mun' => 'Ouro Fino', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '3159605', 'mun' => 'Santa Rita do Sapucaí', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '3165404', 'mun' => 'Sapucaí-Mirim', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '3165404', 'mun' => 'Sapucaí-Mirim', 'uf' => 'MG', 'url' => ''],
    ['ibge' => '5000609', 'mun' => 'Amambai', 'uf' => 'MS', 'url' => ''],
    ['ibge' => '5002951', 'mun' => 'Chapadão do Sul', 'uf' => 'MS', 'url' => ''],
    ['ibge' => '5003207', 'mun' => 'Corumbá', 'uf' => 'MS', 'url' => 'http://nfse.corumba.ms.gov.br:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '5003488', 'mun' => 'Dois Irmãos do Buriti', 'uf' => 'MS', 'url' => ''],
    ['ibge' => '5006002', 'mun' => 'Nova Alvorada do Sul', 'uf' => 'MS', 'url' => ''],
    ['ibge' => '5006309', 'mun' => 'Paranaíba', 'uf' => 'MS', 'url' => ''],
    ['ibge' => '5101803', 'mun' => 'Barra do Garças', 'uf' => 'MT', 'url' => ''],
    ['ibge' => '5101902', 'mun' => 'Brasnorte', 'uf' => 'MT', 'url' => ''],
    ['ibge' => '5104807', 'mun' => 'Jaciara', 'uf' => 'MT', 'url' => 'http://intranet.jaciara.mt.gov.br:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '5105234', 'mun' => 'Lambari D Oeste', 'uf' => 'MT', 'url' => ''],
    ['ibge' => '5105622', 'mun' => 'Mirassol D Oeste', 'uf' => 'MT', 'url' => 'http://186.235.81.12:5663/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '1502152', 'mun' => 'Canaã dos Carajás', 'uf' => 'PA', 'url' => ''],
    ['ibge' => '1503606', 'mun' => 'Itaituba', 'uf' => 'PA', 'url' => ''],
    ['ibge' => '1506161', 'mun' => 'Rio Maria', 'uf' => 'PA', 'url' => 'http://177.75.236.132:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '1508407', 'mun' => 'Xinguara', 'uf' => 'PA', 'url' => ''],
    ['ibge' => '2203909', 'mun' => 'Floriano', 'uf' => 'PI', 'url' => ''],
    ['ibge' => '2207702', 'mun' => 'Parnaíba', 'uf' => 'PI', 'url' => ''],
    ['ibge' => '2208007', 'mun' => 'Picos', 'uf' => 'PI', 'url' => ''],
    ['ibge' => '3301157', 'mun' => 'Cardoso Moreira', 'uf' => 'RJ', 'url' => ''],
    ['ibge' => '1100023', 'mun' => 'Ariquemes', 'uf' => 'RO', 'url' => 'http://170.82.43.84:5660/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '1100098', 'mun' => 'Espigão D´Oeste', 'uf' => 'RO', 'url' => ''],
    ['ibge' => '1100114', 'mun' => 'Jaru', 'uf' => 'RO', 'url' => ''],
    ['ibge' => '1100122', 'mun' => 'Ji-Paraná', 'uf' => 'RO', 'url' => 'http://177.124.184.59:5660/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '1100130', 'mun' => 'Machadinho D´Oeste', 'uf' => 'RO', 'url' => ''],
    ['ibge' => '1100155', 'mun' => 'Ouro Preto do Oeste', 'uf' => 'RO', 'url' => ''],
    ['ibge' => '4315313', 'mun' => 'Quatro Irmãos', 'uf' => 'RS', 'url' => ''],
    ['ibge' => '4311809', 'mun' => 'Marau', 'uf' => 'RS', 'url' => 'http://nfse.pmmarau.com.br:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '4320404', 'mun' => 'Serafina Corrêa', 'uf' => 'RS', 'url' => ''],
    ['ibge' => '4321634', 'mun' => 'Tres Arroios', 'uf' => 'RS', 'url' => 'http://portal.pmtresarroios.com.br:8888/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3501004', 'mun' => 'Altinópolis', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3501509', 'mun' => 'Alvinlândia', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3501806', 'mun' => 'Américo de Campos', 'uf' => 'SP', 'url' => 'http://189.108.31.114:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3502903', 'mun' => 'Araçoiaba da Serra', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3504008', 'mun' => 'Assis', 'uf' => 'SP', 'url' => 'https://nfsews.assis.sp.gov.br/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3504602', 'mun' => 'Bady Bassitt', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3504800', 'mun' => 'Bálsamo', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3505203', 'mun' => 'Bariri', 'uf' => 'SP', 'url' => 'http://sipweb.bariri.sp.gov.br:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3507803', 'mun' => 'Brodowski', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3508306', 'mun' => 'Cabrália Paulista', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3509304', 'mun' => 'Cajobi', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3510104', 'mun' => 'Cândido Rodrigues', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3510302', 'mun' => 'Capela do Alto', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3511300', 'mun' => 'Cedral', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3512100', 'mun' => 'Colômbia', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3512902', 'mun' => 'Cosmorama', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3514403', 'mun' => 'Dracena', 'uf' => 'SP', 'url' => 'http://187.17.201.221:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3514502', 'mun' => 'Duartina', 'uf' => 'SP', 'url' => 'http://186.224.0.62:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3515608', 'mun' => 'Fernando Prestes', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3515509', 'mun' => 'Fernandópolis', 'uf' => 'SP', 'url' => 'http://servicos.fernandopolis.sp.gov.br:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3516705', 'mun' => 'Garça', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3517208', 'mun' => 'Guaiçara', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3517307', 'mun' => 'Guaimbê', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3517406', 'mun' => 'Guaíra', 'uf' => 'SP', 'url' => 'http://186.211.16.210:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3517703', 'mun' => 'Guará', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3518206', 'mun' => 'Guararapes', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3519055', 'mun' => 'Holambra', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3519303', 'mun' => 'Ibaté', 'uf' => 'SP', 'url' => 'http://200.205.131.110:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3519600', 'mun' => 'Ibitinga', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3520103', 'mun' => 'Igarapava', 'uf' => 'SP', 'url' => 'http://138.117.189.204:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3520707', 'mun' => 'Indiaporã', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3520806', 'mun' => 'Inúbia Paulista', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3521903', 'mun' => 'Itajobi', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3525706', 'mun' => 'José Bonifácio', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3526001', 'mun' => 'Junqueirópolis', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3526407', 'mun' => 'Laranjal Paulista', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3529203', 'mun' => 'Martinópolis', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3529708', 'mun' => 'Miguelópolis', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3530102', 'mun' => 'Mirandópolis', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3530409', 'mun' => 'Mirassolândia', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3531100', 'mun' => 'Mongaguá', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3531407', 'mun' => 'Monte Aprazível', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3532009', 'mun' => 'Morungaba', 'uf' => 'SP', 'url' => 'http://servicos.morungaba.sp.gov.br:5600/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3532801', 'mun' => 'Nova Aliança', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3533007', 'mun' => 'Nova Granada', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3533502', 'mun' => 'Novo Horizonte', 'uf' => 'SP', 'url' => 'http://nfews.novohorizonte.sp.gov.br/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3535101', 'mun' => 'Palmares Paulista', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3535408', 'mun' => 'Panorama', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3536307', 'mun' => 'Patrocínio Paulista', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3536604', 'mun' => 'Paulo de Faria', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3537107', 'mun' => 'Pedreira', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3538907', 'mun' => 'Pirajuí', 'uf' => 'SP', 'url' => 'http://189.57.88.130:3392/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3539004', 'mun' => 'Pirangi', 'uf' => 'SP', 'url' => 'http://portal.pmpirangi.com.br:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3539202', 'mun' => 'Pirapozinho', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3540200', 'mun' => 'Pontal', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3540804', 'mun' => 'Potirendaba', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3541604', 'mun' => 'Promissão', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3541703', 'mun' => 'Quatá', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3541802', 'mun' => 'Queiroz', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3542909', 'mun' => 'Ribeirão Bonito', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3543105', 'mun' => 'Ribeirão Corrente', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3544004', 'mun' => 'Rio das Pedras', 'uf' => 'SP', 'url' => 'http://riodaspedras.brgov.com.br:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3544202', 'mun' => 'Riolândia', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3544608', 'mun' => 'Sabino', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3544806', 'mun' => 'Sales', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3544905', 'mun' => 'Sales Oliveira', 'uf' => 'SP', 'url' => 'http://issweb.salesoliveira.sp.gov.br:9090/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3545308', 'mun' => 'Salto de Pirapora', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3547700', 'mun' => 'Santo Anastácio', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3548005', 'mun' => 'Santo Antônio de Posse', 'uf' => 'SP', 'url' => 'http://177.36.128.42:8080/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3549409', 'mun' => 'São Joaquim da Barra', 'uf' => 'SP', 'url' => 'http://187.72.128.113:5661/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3551405', 'mun' => 'Serra Azul', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3551603', 'mun' => 'Serra Negra', 'uf' => 'SP', 'url' => 'http://intra.serranegra.sp.gov.br/IssWeb-ejb/IssWebWS/IssWebWS'],
    ['ibge' => '3552304', 'mun' => 'Sud Mennucci', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3555356', 'mun' => 'Ubarana', 'uf' => 'SP', 'url' => ''],
    ['ibge' => '3555802', 'mun' => 'Urânia', 'uf' => 'SP', 'url' => ''],
];

$am = [];
foreach ($muns as $m) {
    $am["{$m['ibge']}"] = [
        "municipio"   => $m['mun'],
        "uf"          => $m['uf'],
        "homologacao" => $urlhom,
        "producao"    => $m['url'],
        "version"     => "2.01",
        "msgns"       => $ns,
        "soapns"      => $soapns
    ];
}

file_put_contents('../storage/urls_webservices.json', json_encode($am, JSON_PRETTY_PRINT));
